File: project/app/src/main/java/org/owntracks/android/injection/modules/ActivityModules/MapActivityModule.java
package org.owntracks.android.injection.modules.ActivityModules;

import android.support.v7.app.AppCompatActivity;

import org.owntracks.android.injection.modules.ActivityModule;
import org.owntracks.android.injection.qualifier.ActivityContext;
import org.owntracks.android.injection.scopes.PerActivity;
import org.owntracks.android.ui.map.MapActivity;
import org.owntracks.android.ui.map.MapMvvm;
import org.owntracks.android.ui.map.MapViewModel;

import dagger.Binds;
import dagger.Module;

@Module(includes = ActivityModule.class)
public abstract class MapActivityModule {

    @Binds
    @PerActivity
    abstract AppCompatActivity appCompatActivity(MapActivity a);

    @Binds abstract MapMvvm.ViewModel bindViewModel(MapViewModel viewModel);
}

File: project/app/src/main/java/org/owntracks/android/ui/map/MapActivityModule.kt
package org.owntracks.android.ui.map

import dagger.Binds
import dagger.Module
import dagger.hilt.InstallIn
import dagger.hilt.android.components.ActivityComponent
import dagger.hilt.android.scopes.ActivityScoped

@InstallIn(ActivityComponent::class)
@Module
abstract class MapActivityModule {
    @Binds
    @ActivityScoped
    abstract fun bindViewModel(viewModel: MapViewModel): MapMvvm.ViewModel<MapMvvm.View>
}
